dotCMS/src/main/java/com/dotcms/contenttype/business/sql/ContentTypeSql.java

```diff
@@ -15,76 +15,84 @@ public abstract class ContentTypeSql {
 
 	private static ContentTypeSql instance;
 
-    public static ContentTypeSql getInstance() {
-        if (instance == null){
-            instance = DbConnectionFactory.isMySql() ? new ContentTypeMySql() :
-                    DbConnectionFactory.isPostgres() ? new ContentTypePostgres()
-                        : DbConnectionFactory.isMsSql() ? new ContentTypeMSSQL() :
-                            new ContentTypeOracle();
-        }
-        return instance;
-    }
-
-    public static String SELECT_ALL_STRUCTURE_FIELDS = "select  inode.inode as inode, owner, idate as idate, name, "
-        + "description, default_structure, page_detail, structuretype, system, fixed, velocity_var_name , "
-        + "url_map_pattern , host, folder, expire_date_var , publish_date_var , mod_date, icon, sort_order "
-        + "from inode, structure  where inode.type='structure' and inode.inode = structure.inode  ";
-    
-    public static String SELECT_ONLY_INODE_FIELD = "select  inode.inode as inode from inode, structure  where inode.type='structure' and inode.inode = structure.inode  ";
+	public static ContentTypeSql getInstance() {
+		if (instance == null){
+			instance = DbConnectionFactory.isMySql() ? new ContentTypeMySql() :
```

Comment:
I think you can reduce it to PG and MS

Reply:
I plan to work on this in a separate PR sometime in the future. I really dislike what was done here to "instantiate" different interfaces modifying their attributes with db specific SQL